### ISSUE TITLE
Wazuh active resposne concatenating args

### ISSUE THREAD

**Issue by JGRFenix** [wazuh/wazuh#7521]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.13.1 | Active Response | Agent (I think, could be both) | Wazuh Repo | Windows |

Hiya all, So when building my own Active response script for Windows I've noticed some strange behaviour. 

Wazuh seems to concatenate the args sent to it via api if theres more than 3 EG: 
When sending this json:
```json
{
  "command": "win-example_script0",
  "arguments": [
    "example_1",
    "example_2",
    "example_3",
    "example_4",
    "example_5"		
  ]
}
```
Against this script:
```cmd
@ECHO OFF
ECHO. 

:: Logging it all
FOR /F "TOKENS=1* DELIMS= " %%A IN ('DATE/T') DO SET DATE=%%B
FOR /F "TOKENS=1* DELIMS= " %%A IN ('TIME/T') DO SET TIME=%%A
ECHO %DATE% %TIME% %0 %1 %2 %3 %4 %5 %6 %7 %8 %9 >> active-response/active-responses.log
ECHO %DATE% %TIME% %0 %4  >> active-response/active-responses.log
```
it will produce these logs:
```
17/02/2021 10:08 "active-response/bin/ar_win_example_script.cmd" add "example_1" "example_2" "example_3 example_4 example_5"

17/02/2021 10:08 "active-response/bin/ar_win_example_script.cmd" "example_3 example_4 example_5"
```
From reading the documentation https://documentation.wazuh.com/3.13/user-manual/capabilities/active-response/remediation-faq.html#can-i-use-a-custom-script-for-active-responses it looks like theres support for up to 6 args that can be passed through? 

I also tried using https://documentation.wazuh.com/3.13/user-manual/reference/ossec-conf/commands.html?highlight=command#extra-args however this doesn't seem to work when calling the active response from the API at all

**Comment by jnasselle** [comment]
Hi @JGRFenix !

I'm trying to reproduce your problem in order check what's going on. In the meantime, could you please share to me you `active-response` and `command` configuration sections?

Regards,
Nico

**Comment by JGRFenix** [comment]
Hi @jnasselle 

Thanks for looking into this for me. 

Of course. Here is the info from the ossec.conf

command
```
<command>
    <name>win-example_script</name>
    <executable>ar_win_example_script.cmd</executable>
    <extra_args>example_3 example_4 example_5</extra_args>
  </command>
```
active-response:
```
   <active-response>
    <command>win-example_script</command>
    <location>local</location>
  </active-response>
```

Please let me know if you have any other questions

Thanks,

JR

**Comment by TomasTurina** [comment]
Hi @JGRFenix,

I've been looking at the [code](https://github.com/wazuh/wazuh/blob/3.13/src/os_execd/win_execd.c) that creates and sends the message to the Windows scripts.

To understand what is happening, let see this line:

> snprintf(buffer, OS_MAXSTR, name[0] == '!' ? "\"%s\" %s %s %s %s" : "\"%s\" %s \"%s\" \"%s\" \"%s\"", command, ADD_ENTRY, cmd_user, cmd_ip, tmp_msg);

As you can see:

- The first argument is expected to be a user name.
- The second argument is expected to be an IP.
- The rest of the arguments are sent to the script all together.

This is the behavior for Windows, which is different to Linux behavior where all arguments are sent separately.

We have recently changed the whole behavior of active response, and now the string arguments will be replaced by a JSON message which will be send completely to the active response script. [Here](https://github.com/wazuh/wazuh/issues/6954) you can find more information about this.

I hope you find this information helpful.

**Comment by JGRFenix** [comment]
Hi @TomasTurina 

Thanks for explaining this. 

I like that Active response is moving to JSON input thats awesome! 

Will this be the same for both Linux and Windows?

**Comment by TomasTurina** [comment]
Hi @JGRFenix,

Yes, it will be the same for all the OS systems that Wazuh supports.

It is already implemented and it will be available in the next release of Wazuh (v4.2.0).

Best regards.
